### PR TITLE
Improve snap building process

### PR DESCRIPTION
**Pull request by kissiel**
This PR rearranges the snaps' code, so:

* it's possible to build snaps for all releases
* common bits are reused
* there's no need for per-series branches

### CODE REVIEW

**Comment by kissiel** [comment]
Some of the diffs are rendered as changing series, but that's just an artifact from git. Follow paths to see real changes.

**Review by yphus — CHANGES_REQUESTED** [review]
Checkbox core snaps are not using local sources, instead they keep loading parts source from launchpad:

e.g:
```
  checkbox-ng:
    source-tag: ""
    source-depth: 1
    plugin: python
    source: https://git.launchpad.net/checkbox-ng
    source-type: git
```
must be replaced with:
```
  checkbox-ng:
    plugin: python
    source: ../../checkbox-ng
```

**Review by yphus — CHANGES_REQUESTED** [review]
The slots definition have to be updated to match the new provider names:

e.g:
```
slots:
  provider-resource:
    interface: content
    read:
      - $SNAP/providers/plainbox-provider-resource
  provider-checkbox:
    interface: content
    read:
      - $SNAP/providers/plainbox-provider-checkbox
```
must be replaced with:
```
slots:
  provider-resource:
    interface: content
    read:
      - $SNAP/providers/checkbox-provider-resource
  provider-checkbox:
    interface: content
    read:
      - $SNAP/providers/checkbox-provider-base
```

Warning: Do **not** change the slot name (provider-checkbox -> provider-base) in order to stay compatible with all checkbox-<project> snaps

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+
+base: core
+
+# Don't forget to add a new slot if a new provider part is added in the parts
+# section below.
+slots:
+  provider-resource:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-resource
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:17]:
```suggestion
      - $SNAP/providers/checkbox-provider-resource
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+  provider-checkbox:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-checkbox
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:21]:
```suggestion
      - $SNAP/providers/checkbox-provider-checkbox
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+  provider-ipdt:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-ipdt
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:25]:
```suggestion
      - $SNAP/providers/checkbox-provider-ipdt
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+  provider-docker:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-docker
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:29]:
```suggestion
      - $SNAP/providers/checkbox-provider-docker
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+  provider-tpm2:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-tpm2
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:33]:
```suggestion
      - $SNAP/providers/checkbox-provider-tpm2
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 6]
```diff
@@ -0,0 +1,435 @@
...
+  provider-engineering-tests:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-engineering-tests
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:37]:
```suggestion
      - $SNAP/providers/checkbox-provider-engineering-tests
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 7]
```diff
@@ -0,0 +1,435 @@
...
+  provider-sru:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-sru
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:41]:
```suggestion
      - $SNAP/providers/checkbox-provider-sru
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 8]
```diff
@@ -0,0 +1,435 @@
...
+    read:
+      - $SNAP/providers/checkbox-provider-edgex
+  provider-gpgpu:
+    interface: content
+    read:
+      - $SNAP/providers/checkbox-provider-gpgpu
+  provider-certification-client:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-certification-client
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:53]:
```suggestion
      - $SNAP/providers/checkbox-provider-certification-client
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 9]
```diff
@@ -0,0 +1,435 @@
...
+  provider-certification-server:
+    interface: content
+    read:
+      - $SNAP/providers/plainbox-provider-certification-server
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:57]:
```suggestion
      - $SNAP/providers/checkbox-provider-certification-server
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 10]
```diff
@@ -0,0 +1,435 @@
...
+    override-pull: |
+      PYTHONHOME=/usr PYTHONUSERBASE=$SNAPCRAFT_PART_INSTALL $SNAPCRAFT_PART_INSTALL/usr/bin/python3 -m pip install --upgrade 'pip; python_version >= "3.6"' 'pip<21; python_version < "3.6"' --user
+      snapcraftctl pull
+################################################################################
+  checkbox-ng:
+    source-tag: ""
+    source-depth: 1
+    plugin: python
+    source: https://git.launchpad.net/checkbox-ng
+    source-type: git
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:162]:
```suggestion
  checkbox-ng:
    plugin: python
    source: checkbox-ng
    source-type: local
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 11]
```diff
@@ -0,0 +1,435 @@
...
+      - python3-urwid
+      - python3-xlsxwriter
+      - python3-setuptools
+    python-packages:
+      - tqdm
+    after: [checkbox-support]
+    build-environment:
+      - PYTHONPATH: $SNAPCRAFT_PART_INSTALL/usr/lib/python3/dist-packages:$PYTHONPATH
+################################################################################
+  plainbox-provider-resource:
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:180]:
```suggestion
  checkbox-provider-resource:
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 12]
```diff
@@ -0,0 +1,435 @@
...
+      - pciutils
+      - python3-requests-unixsocket
+      - smartmontools
+    override-build: |
+      cd src && autoreconf -i && cd -
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-resource --root="$SNAPCRAFT_PART_INSTALL"
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:198]:
```suggestion
      python3 manage.py install --layout=relocatable --prefix=/providers/checkbox-provider-resource --root="$SNAPCRAFT_PART_INSTALL"
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 13]
```diff
@@ -0,0 +1,435 @@
...
+  plainbox-provider-checkbox:
+    plugin: dump
+    source: providers/base
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-checkbox --root="$SNAPCRAFT_PART_INSTALL"
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:216]:
```suggestion
      python3 manage.py install --layout=relocatable --prefix=/providers/checkbox-provider-base --root="$SNAPCRAFT_PART_INSTALL"
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 14]
```diff
@@ -0,0 +1,435 @@
...
+  plainbox-provider-ipdt:
+    plugin: dump
+    source: providers/ipdt
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-ipdt --root="$SNAPCRAFT_PART_INSTALL"
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:295]:
```suggestion
      python3 manage.py install --layout=relocatable --prefix=/providers/checkbox-provider-ipdt --root="$SNAPCRAFT_PART_INSTALL"
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 15]
```diff
@@ -0,0 +1,435 @@
...
+    stage-packages:
+      - util-linux
+    after: [plainbox-provider-checkbox]
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:298]:
```suggestion
    after: [checkbox-provider-base]
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 16]
```diff
@@ -0,0 +1,435 @@
...
+################################################################################
+  plainbox-provider-docker:
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:300]:
```suggestion
  checkbox-provider-docker:
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 17]
```diff
@@ -0,0 +1,435 @@
...
+    plugin: dump
+    source: providers/docker
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-docker --root="$SNAPCRAFT_PART_INSTALL"
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:309]:
```suggestion
      python3 manage.py install --layout=relocatable --prefix=/providers/checkbox-provider-docker --root="$SNAPCRAFT_PART_INSTALL"
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 18]
```diff
@@ -0,0 +1,435 @@
...
+    stage-packages:
+      - apache2-utils
+    after: [plainbox-provider-ipdt]
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:312]:
```suggestion
    after: [checkbox-provider-ipdt]
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 19]
```diff
@@ -0,0 +1,435 @@
...
+  plainbox-provider-tpm2:
+    plugin: dump
+    source: providers/tpm2
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-tpm2 --root="$SNAPCRAFT_PART_INSTALL"
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:323]:
```suggestion
      python3 manage.py install --layout=relocatable --prefix=/providers/checkbox-provider-tpm2 --root="$SNAPCRAFT_PART_INSTALL"
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 20]
```diff
@@ -0,0 +1,435 @@
...
+    after: [plainbox-provider-docker]
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:324]:
```suggestion
    after: [checkbox-provider-docker]
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 21]
```diff
@@ -0,0 +1,435 @@
...
+################################################################################
+  plainbox-provider-sru:
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:326]:
```suggestion
  checkbox-provider-sru:
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 22]
```diff
@@ -0,0 +1,435 @@
...
+    plugin: dump
+    source: providers/sru
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-sru --root="$SNAPCRAFT_PART_INSTALL"
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:335]:
```suggestion
      python3 manage.py install --layout=relocatable --prefix=/providers/checkbox-provider-sru --root="$SNAPCRAFT_PART_INSTALL"
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 23]
```diff
@@ -0,0 +1,435 @@
...
+    after: [plainbox-provider-tpm2]
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:336]:
```suggestion
    after: [checkbox-provider-tpm2]
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 24]
```diff
@@ -0,0 +1,435 @@
...
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-edgex --root="$SNAPCRAFT_PART_INSTALL"
+    stage-packages:
+      - curl
+    after: [plainbox-provider-sru]
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:350]:
```suggestion
    after: [checkbox-provider-sru]
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 25]
```diff
@@ -0,0 +1,435 @@
...
+  checkbox-provider-gpgpu:
+    plugin: dump
+    source: providers/gpgpu
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-gpgpu --root="$SNAPCRAFT_PART_INSTALL"
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:361]:
```suggestion
      python3 manage.py install --layout=relocatable --prefix=/providers/checkbox-provider-gpgpu --root="$SNAPCRAFT_PART_INSTALL"
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 26]
```diff
@@ -0,0 +1,435 @@
...
+    after: [checkbox-provider-edgex]
+################################################################################
+  plainbox-provider-certification-client:
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:364]:
```suggestion
  checkbox-provider-certification-client:
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 27]
```diff
@@ -0,0 +1,435 @@
...
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-certification-client --root="$SNAPCRAFT_PART_INSTALL"
+    after: [checkbox-provider-gpgpu]
+################################################################################
+  plainbox-provider-certification-server:
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:376]:
```suggestion
  checkbox-provider-certification-server:
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 28]
```diff
@@ -0,0 +1,435 @@
...
+    plugin: dump
+    source: providers/certification-server
+    source-type: local
+    override-build: |
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.8/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
+      python3 manage.py validate
+      python3 manage.py build
+      python3 manage.py install --layout=relocatable --prefix=/providers/plainbox-provider-certification-server --root="$SNAPCRAFT_PART_INSTALL"
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:385]:
```suggestion
      python3 manage.py install --layout=relocatable --prefix=/providers/checkbox-provider-certification-server --root="$SNAPCRAFT_PART_INSTALL"
```

File: checkbox-core-snap/series16/snap/snapcraft.yaml [hunk 29]
```diff
@@ -0,0 +1,435 @@
...
+      for p in `ls -d ../../*`; do
+          if [ -d $p/src ]; then
+            (cd $p/src
+                if [ -d $p/src/.git ]; then
+                    (echo `basename $p`\: && git show --pretty=format:"%h%d %aN %ci%n%s%n" -q ; echo ) >> $SNAPCRAFT_PART_INSTALL/parts_meta_info
+                fi # for additional `source-type` support, elif them here
+            )
+          fi
+      done
+    after: [plainbox-provider-certification-server]
```
Comment (yphus) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:431]:
```suggestion
    after: [checkbox-provider-certification-server]
```

**Review by yphus — CHANGES_REQUESTED** [review]
Providers name have changed and all parts have to be updated to match. I've suggested edits to s16 but the same fixes are required for s18/20 and 22.

Moreover checkbox-ng and checkbox-supprt are still using lp a sources

File: checkbox-snap/series_uc22/snap/snapcraft.yaml [hunk 30]
```diff
@@ -0,0 +1,102 @@
...
+base: core22
+
+plugs:
+  checkbox-runtime:
+    interface: content
+    target: $SNAP/checkbox-runtime
+    default-provider: checkbox22
+  provider-resource:
+    interface: content
+    target: $SNAP/providers/plainbox-provider-resource
```
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:19]:
```suggestion
    target: $SNAP/providers/checkbox-provider-resource
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml [hunk 31]
```diff
@@ -0,0 +1,102 @@
...
+    default-provider: checkbox22
+  provider-checkbox:
+    interface: content
+    target: $SNAP/providers/plainbox-provider-checkbox
```
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:23]:
```suggestion
    target: $SNAP/providers/checkbox-provider-base
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml [hunk 32]
```diff
@@ -0,0 +1,102 @@
...
+    default-provider: checkbox22
+  provider-ipdt:
+    interface: content
+    target: $SNAP/providers/plainbox-provider-ipdt
```
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:27]:
```suggestion
    target: $SNAP/providers/checkbox-provider-ipdt
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml (hunk 30 above)
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:19]:
```suggestion
    target: $SNAP/providers/checkbox-provider-resource
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml (hunk 31 above)
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:23]:
```suggestion
    target: $SNAP/providers/checkbox-provider-base
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml (hunk 32 above)
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:27]:
```suggestion
    target: $SNAP/providers/checkbox-provider-ipdt
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml [hunk 33]
```diff
@@ -0,0 +1,102 @@
...
+    default-provider: checkbox22
+  provider-docker:
+    interface: content
+    target: $SNAP/providers/plainbox-provider-docker
```
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:31]:
```suggestion
    target: $SNAP/providers/checkbox-provider-docker
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml [hunk 34]
```diff
@@ -0,0 +1,102 @@
...
+    default-provider: checkbox22
+  provider-tpm2:
+    interface: content
+    target: $SNAP/providers/plainbox-provider-tpm2
+    default-provider: checkbox22
+  provider-sru:
+    interface: content
+    target: $SNAP/providers/plainbox-provider-sru
```
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:39]:
```suggestion
    target: $SNAP/providers/checkbox-provider-sru
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml [hunk 35]
```diff
@@ -0,0 +1,102 @@
...
+    default-provider: checkbox22
+  provider-gpgpu:
+    interface: content
+    target: $SNAP/providers/checkbox-provider-gpgpu
+    default-provider: checkbox22
+  provider-certification-client:
+    interface: content
+    target: $SNAP/providers/plainbox-provider-certification-client
```
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:47]:
```suggestion
    target: $SNAP/providers/checkbox-provider-certification-client
```

File: checkbox-snap/series_uc22/snap/snapcraft.yaml [hunk 36]
```diff
@@ -0,0 +1,102 @@
...
+    default-provider: checkbox22
+  provider-certification-server:
+    interface: content
+    target: $SNAP/providers/plainbox-provider-certification-server
```
Comment (yphus) [inline checkbox-snap/series_uc22/snap/snapcraft.yaml:51]:
```suggestion
    target: $SNAP/providers/checkbox-provider-certification-server
```

**Review by yphus — CHANGES_REQUESTED** [review]
All the checkbox snaps also can get rid of plainbox prefixes. I've suggested edits for series_uc22 but all content interfaces based snaps are impacted (uc16,18,20 and 22)

**Comment by kissiel** [comment]
I applied a lot of suggested changes. I'll do more testing, but I think it's ready for another look-through.

File: checkbox-core-snap/series22/snap/snapcraft.yaml [hunk 37]
```diff
@@ -229,7 +228,11 @@ parts:
       - smartmontools
     override-build: |
       cd src && autoreconf -i && cd -
-      snapcraftctl build
+      export PYTHONPATH=$SNAPCRAFT_STAGE/lib/python3.10/site-packages:$SNAPCRAFT_STAGE/usr/lib/python3/dist-packages
+      for path in $(find "$SNAPCRAFT_STAGE/providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
```
Comment (yphus) [inline checkbox-core-snap/series22/snap/snapcraft.yaml:232]:
```suggestion
      for path in $(find "$SNAPCRAFT_STAGE/../providers/" -mindepth 1 -maxdepth 1 -type d); do export PROVIDERPATH=$path${PROVIDERPATH:+:$PROVIDERPATH}; done
```

Just for the first provider as the stage area is not yet populated; the PROVIDERPATH env var would then be empty and system providers would conflicts during a local build

**Review by yphus — CHANGES_REQUESTED** [review]
`s/checkbox-provider-checkbox/checkbox-provider-base/g`

Add tag versions for fwts and acpi-tools:
```
  fwts:
    source-tag: "V22.09.00"
```
[...]
```
  acpi-tools:
    source-tag: "R03_31_22"
```

**Review by pieqq — APPROVED** [review]
LGTM!

**Review by yphus — APPROVED** [review]
Thanks, LGTM too

**Pull request merged**